ethereum/dex/liquidity/insert_balancer_v1.sql

```diff
@@ -0,0 +1,198 @@
+CREATE OR REPLACE FUNCTION dex.insert_liquidity_balancer_v1(start_ts timestamptz, end_ts timestamptz=now()) RETURNS integer
+LANGUAGE plpgsql AS $function$
+DECLARE r integer;
+BEGIN
+WITH days as ( -- update table entries until previous day
+    SELECT day FROM generate_series(start_ts, (SELECT end_ts - interval '1 day'), '1 day') g(day)
+), 
+pools_v1 AS (
+    SELECT pool
+    FROM balancer."BFactory_evt_LOG_NEW_POOL"
+),
+pools_tokens_v1 AS ( -- 15243 rows
+    SELECT DISTINCT
+    p.pool,
+    e.contract_address AS token
+    FROM erc20."ERC20_evt_Transfer" e
+    INNER JOIN pools_v1 p ON e."to" = p.pool
+    -- AND e.evt_block_time >= start_ts
```

Comment:
This is not needed I assume because we want all tokens and the args are used below. Drop these 2 lines?

Reply:
speaking of liquidity I don't even think you need all tokens, but only those which are part of the pool. Looking at all token transfers you'll probably find tokens that were sent to the pool by mistake. For this reason, I would change this CTE to:

```
pools_tokens_v1 AS (
    select contract_address as pool, token
    from balancer."BPool_call_bind"
    where call_success
)
```